components/dropdown/dropdown-content-mixin.js

```diff
@@ -0,0 +1,556 @@
+		document.body.addEventListener('click', this.__onAutoCloseClick, true);
+
+		this.addEventListener('d2l-dropdown-close', this.__onClose);
+		this.addEventListener('d2l-dropdown-position', this.__toggleScrollStyles);
+	}
+
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		window.removeEventListener('resize', this.__onResize);
+		document.body.removeEventListener('focus', this.__onAutoCloseFocus, true);
+		document.body.removeEventListener('click', this.__onAutoCloseClick, true);
+		this.removeEventListener('d2l-dropdown-close', this.__onClose);
+		this.removeEventListener('d2l-dropdown-position', this.__toggleScrollStyles);
+		this.__content.removeEventListener('scroll', this.__toggleScrollStyles);
+		clearDismissible(this.__dismissibleId);
+		this.__dismissibleId = null;
+	}
+
+	updated(changedProperties) {
+		changedProperties.forEach((_, propName) => {
+			if (propName === 'verticalOffset') {
+				// for IE11
+				if (window.ShadyCSS) window.ShadyCSS.styleSubtree(this, { '--d2l-dropdown-verticaloffset': `${this.verticalOffset}px` });
+				else this.style.setProperty('--d2l-dropdown-verticaloffset', `${this.verticalOffset}px`);
+			}
+		});
+	}
+
+	_renderContentContainer(innerHtml) {
+		const styles = {};
+		if (this.minWidth) {
+			styles.minWidth = `${this.minWidth}px`;
+		}
+
+		return html`
+			<div class="d2l-dropdown-content-container" style=${styleMap(styles)}>
+				${innerHtml()}
+			</div>
+		`;
+	}
+
+	_renderWidthContainer(innerHtml) {
+		const styles = {};
+		if (this.maxWidth) {
+			styles.maxWidth = `${this.maxWidth}px`;
+		}
+		if (this.minWidth) {
+			styles.minWidth = `${this.minWidth}px`;
+		}
+
+		return html`
+			<div class="d2l-dropdown-content-width" style=${styleMap(styles)}>
+				${innerHtml()}
+			</div>
+		`;
+	}
+
+	/**
+	 * Closes/hides the dropdown.
+	 */
+	close() {
+		this.opened = false;
+	}
+
+	/**
+	 * Opens/shows the dropdown.
+	 * @param {Boolean} applyFocus Whether focus should be automatically move to first focusable upon opening.
+	 */
+	open(applyFocus) {
+		this.__applyFocus = applyFocus !== undefined ? applyFocus : true;
+		this.opened = true;
+	}
+
+	/**
+	 * Synchronously stamps and attaches the content into the DOM. By default, rendering of the
+	 * content into the DOM is deferred as a performance optimization, so if access to the content
+	 * DOM is required (for example by calling `document.querySelector`) before opening the dropdown,
+	 * `forceRender` may be used.
+	 */
+	async forceRender() {
+		if (!this.renderContent) {
+			this.renderContent = true;
+		}
+		await this.updateComplete;
+	}
+
+	/**
+	 * Toggles the opened/closed state of the dropdown.  If closed, it will open, and vice versa.
+	 * @param {Boolean} applyFocus Whether focus should be automatically moved to first focusable upon opening.
+	 */
+	toggleOpen(applyFocus) {
+		if (this.opened) {
+			this.close();
+		} else {
+			this.open(applyFocus);
+		}
+	}
+
+	/**
+	 * Private.
+	 */
+	scrollTo(scrollTop) {

```

Comment:
If these `scrollTo` and `height` methods are really private, we should look at renaming them the be `_scrollTo()` and `height()`.

Reply:
I never actually could find anywhere that these two where being used. The only reason I added them in was due to being nervous that they were being used somewhere in the LMS that I missed and removing them would unintentionally result in broken behavior.

Reply:
Looks like `scrollTo` and `height` are used inside `d2l-time-picker`: https://github.com/BrightspaceUI/time-picker/blob/master/d2l-time-picker.js#L292
I couldn't find any other uses but its possible I missed them since they're difficult functions to search for.